web_extension/content/scripts/new_tab.js

```diff
@@ -14,30 +16,42 @@ function set_image(url) {
 }
 
 function show_info(item) {
-  document.querySelector('info name').appendChild( link(item.external_url, item.content_text) )
-  document.querySelector('info by-line').appendChild( link(item.author.url, item.author.name) )
-  document.querySelector('info venue').appendChild( link(item._meta.venue.url, item._meta.venue.name) )
+  document.querySelector('info name').appendChild(
+    Builder.link(item.external_url, item.content_text)
```

Comment:
I'm highly skeptical but also at this rate you'll have re-invented react or something soon 😂

Reply:
too heavyweight. Unfortunately the `innerHTML = template` thing doesn't work because [it's not safe](https://developer.mozilla.org/en-US/docs/Archive/B2G_OS/Security/Security_Automation) and things